### PR TITLE
feat: support swagger-ui for the openapi

### PR DESCRIPTION
**Pull request by n063h**
ref to #384 

# Description
Remove unnecessary explicit lifetime annotation and refactor signatures of some methods.

# Before
Users can visit /openapi to get the openapi spec of mosec.

# After 
Users can visit /openapi get the openapi spec of mosec, and visit /swagger-ui to access the visual documentation Swagger.
image

### CODE REVIEW

File: src/apidoc.rs [hunk 1]
```diff
@@ -105,9 +80,10 @@ impl MosecOpenAPI {
...
     /// [path_item]: https://spec.openapis.org/oas/latest.html#path-item-object
     /// [utoipa-gen]: https://crates.io/crates/utoipa-gen
-    pub fn replace_path_item(&mut self, from: &str, to: &str) {
-        if let Some(path_item) = self.api.paths.paths.remove(from) {
-            self.api.paths.paths.insert(to.to_owned(), path_item);
+    pub fn replace_path_item(&mut self, from: &str, to: &str) -> &mut Self {
+        if let Some(r) = self.api.paths.paths.remove(from) {
+            self.api.paths.paths.insert(to.to_owned(), r);
         }
+        self
```
Comment (kemingy) [inline src/apidoc.rs:87]: Never been used?

File: src/main.rs [hunk 2]
```diff
@@ -67,22 +68,21 @@ async fn shutdown_signal() {
...
 
     let state = AppState {
         mime: opts.mime.clone(),
-        openapi: api,
     };
     let coordinator = Coordinator::init_from_opts(opts);
     let barrier = coordinator.run();
     barrier.wait().await;
     let app = Router::new()
+        .merge(SwaggerUi::new("/swagger-ui").url("/openapi", doc.api))
```
Comment (kemingy) [inline src/main.rs:84]:
```suggestion
        .merge(SwaggerUi::new("/api/swagger").url("/api/openapi.json", doc.api))
```

File: src/apidoc.rs (hunk 1 above)
Comment (n063h) [reply src/apidoc.rs:87]: To allow method chaining

Comment (n063h) [reply src/apidoc.rs:87]: If there's more than one route to be replaced, it would look cleaner

**Pull request merged**